core/src/main/java/jenkins/model/queue/ProxyTask.java

```diff
@@ -0,0 +1,173 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.model.queue;
+
+import hudson.model.Label;
+import hudson.model.Node;
+import hudson.model.Queue;
+import hudson.model.ResourceList;
+import hudson.model.queue.CauseOfBlockage;
+import hudson.model.queue.SubTask;
+import java.io.IOException;
+import java.util.Collection;
+import javax.annotation.Nonnull;
+import org.acegisecurity.Authentication;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+/**
+ * A {@link Task}, which refers other {@link Task}.
+ * This method allows to proxies all methods of the original {@link Task}.
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+@ExportedBean
+public class ProxyTask implements Queue.Task {
+        
+    private final Queue.Task upperTask;
+
+    /**
+     * Creates a new {@link ProxyTask} for the specified {@link Queue.Task}.
+     * @param task A task to be proxied.
+     */
+    public ProxyTask(@Nonnull Queue.Task task) {
+        this.upperTask = task;
+    }
+
+    /**
+     * Get the task proxied by this class.
+     * @return {@link Task}.
+     */
+    public @Nonnull Queue.Task getUpperTask() {
+        return upperTask;
+    }
+
+    @Exported
+    @Override
+    public boolean isBuildBlocked() {
+        return upperTask.isBuildBlocked();
+    }
+
+    @Exported
+    @Override
+    public String getWhyBlocked() {
+        return upperTask.getWhyBlocked();
+    }
+
+    @Override
+    public CauseOfBlockage getCauseOfBlockage() {
+        return upperTask.getCauseOfBlockage();
+    }
+
+    @Exported
+    @Override
+    public String getName() {
+        return upperTask.getName();
+    }
+
+    @Exported
+    @Override
+    public String getFullDisplayName() {
+        return upperTask.getFullDisplayName();
+    }
+
+    @Override
+    public void checkAbortPermission() {
+        upperTask.checkAbortPermission();
+    }
+
+    @Override
+    public boolean hasAbortPermission() {
+        return upperTask.hasAbortPermission();
+    }
+
+    @Exported
+    @Override
+    public String getUrl() {
+        return upperTask.getUrl();
+    }
+
+    @Exported
+    @Override
+    public boolean isConcurrentBuild() {
+        return upperTask.isConcurrentBuild();
+    }
+
+    @Override
+    public Collection<? extends SubTask> getSubTasks() {
+        return upperTask.getSubTasks();
+    }
+
+    @Override
+    public Authentication getDefaultAuthentication() {
+        return upperTask.getDefaultAuthentication();
+    }
+
+    @Override
+    public Authentication getDefaultAuthentication(Queue.Item item) {
+        return upperTask.getDefaultAuthentication(item);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return upperTask.getDisplayName();
+    }
+
+    @Exported
+    @Override
+    public Label getAssignedLabel() {
```

Comment:
@oleg-nenashev So should #1437 be reworked to also return the label instead of the string for the `AbstractProject`? It makes no real sense to sometimes return one and sometimes the other.